src/index.ts

```diff
@@ -6,7 +6,11 @@ const suspectConstructorRx = /"(?:c|\\u0063)(?:o|\\u006[Ff])(?:n|\\u006[Ee])(?:s
 const JsonSigRx = /^["[{]|^-?\d[\d.]{0,14}$/;
 
 function jsonParseTransform (key: string, value: any): any {
-  if (key === "__proto__" || key === "constructor") {
+  if (key === "__proto__") {
+    return;
+  }
+  if (key === "constructor" && value && typeof value === "object" && ("prototype" in value)) {
```

Comment:
To be honest i'm still worried about performance overhead of this new check. `jsonParseTransform` was a second assurance method in top of regex. But anyway let's have it as hotfix and revise later

Reply:
But that should only bring an overhead for these specific keys, right? 🤔

Reply:
Yes indeed :)